index.d.ts

```diff
@@ -1,67 +1,97 @@
 
-  namespace JwksRsa {
-    class JwksClient {
-      constructor(options: Options);
+    getKeys(cb: (err: Error | null, keys: unknown) => void): void;
+    getSigningKeys(cb: (err: Error | null, keys: SigningKey[]) => void): void;
+    getSigningKey: (kid: string, cb: (err: Error | null, key: SigningKey) => void) => void;
+  }
+
+  interface Headers {
+    [key: string]: string;
+  }
+
+  interface ClientOptions {
+    jwksUri: string;
+    rateLimit?: boolean;
+    cache?: boolean;
+    cacheMaxEntries?: number;
+    cacheMaxAge?: number;
+    jwksRequestsPerMinute?: number;
+    strictSsl?: boolean;
+    requestHeaders?: Headers;
+  }
+
+  interface CertSigningKey {
+    kid: string;
+    nbf: string;
+    publicKey: string;
+  }
+
+  interface RsaSigningKey {
+    kid: string;
+    nbf: string;
+    rsaPublicKey: string;
+  }
+
+  type SigningKey = CertSigningKey | RsaSigningKey;
 
-      getKeys: (cb: (err: Error, keys: Jwk[]) => any) => any;
-      getSigningKeys: (cb: (err: Error, keys: Jwk[]) => any) => any;
-      getSigningKey: (kid: string, cb: (err: Error, key: Jwk) => any) => any;
-    }
+  function expressJwtSecret(options: ExpressJwtOptions): SecretCallbackLong;
 
-    interface Jwk {
-      kid: string;
-      nbf?: number;
-      publicKey?: string;
-      rsaPublicKey?: string;
-    }
+  function passportJwtSecret(options: ExpressJwtOptions): SecretCallback;
 
-    interface Headers {
-      [key: string]: string;
-    }
+  interface ExpressJwtOptions extends ClientOptions {
+    handleSigningKeyError?: (err: Error | null, cb: (err: Error | null) => void) => void;
+  }
 
-    interface Options {
-      jwksUri: string;
-      rateLimit?: boolean;
-      cache?: boolean;
-      cacheMaxEntries?: number;
-      cacheMaxAge?: number;
-      jwksRequestsPerMinute?: number;
-      strictSsl?: boolean;
-      requestHeaders?: Headers;
-      handleSigningKeyError?(err: Error, cb: (err: Error) => void): any;
-    }
+  function hapiJwt2Key(options: HapiJwtOptions): (decodedToken: DecodedToken, cb: HapiCallback) => void;
 
-    function expressJwtSecret(options: JwksRsa.Options): ExpressJwt.SecretCallback;
+  interface HapiJwtOptions extends ClientOptions {
+    handleSigningKeyError?: (err: Error | null, cb: HapiCallback) => void;
+  }
 
-    function hapiJwt2Key(options: JwksRsa.Options): (name: string, scheme: string, options?: any) => void;
+  type HapiCallback = (err: Error | null, publicKey: string, signingKey: SigningKey) => void;
 
-    function hapiJwt2KeyAsync(options: JwksRsa.Options): (name: string, scheme: string, options?: any) => void;
+  interface DecodedToken {
+    header: TokenHeader;
+  }
+
+  interface TokenHeader {
+    alg: string;
+    kid: string;
+  }
 
-    function koaJwtSecret(options: JwksRsa.Options): (name: string, scheme: string, options?: any) => void;
+  function hapiJwt2KeyAsync(options: HapiJwtOptions): (decodedToken: DecodedToken) => Promise<{ key: SigningKey }>;
```

Comment:
```
(decodedToken: DecodedToken) => Promise<{ key: SigningKey }>;
```
this is actually wrong. it does not return a promise of `SigningKey`, they `key` property is just a string with the value. See:
- https://github.com/auth0/node-jwks-rsa/blob/master/src/integrations/hapi.js#L25
- https://github.com/auth0/node-jwks-rsa/blob/master/src/integrations/hapi.js#L58

I wrote a test for verifying the typings here https://github.com/auth0/node-jwks-rsa/pull/98

Reply:
Good catch, thanks! Fixed it.